llama.h

```diff
@@ -111,6 +111,15 @@ extern "C" {
         LLAMA_ROPE_SCALING_MAX_VALUE   = LLAMA_ROPE_SCALING_YARN,
     };
 
+    enum llama_numa_strategies {
+        LLAMA_NUMA_STRATEGY_DISABLED = 0,
+        LLAMA_NUMA_STRATEGY_INTERLEAVE = 1,
+        LLAMA_NUMA_STRATEGY_ISOLATE = 2,
+        LLAMA_NUMA_STRATEGY_NUMACTL = 3,
+        LLAMA_NUMA_STRATEGY_MIRROR = 4,
+        LLAMA_NUMA_STRATEGY_MAX_VALUE = LLAMA_NUMA_STRATEGY_MIRROR,
+    };
```

Comment:
I am not sure if @ggerganov agrees, but to me it seems unnecessary to duplicate this enum here, since `llama.h` already includes `ggml.h`.

Reply:
@slaren  It comes down to llama.h not being included in ggml and ggml.h not being included in common.h. Is there a good way to share the same enum across both worlds?

Reply:
I think it would be reasonable to include ggml.h in common.h, and use ggml_numa_strategies everywhere.

Reply:
That would simplify things greatly if its ok 
Any objection @ggerganov?

Reply:
I made this change locally on my branch and it has indeed made things much cleaner. I've also cleaned up the "examples" that were broken by the change from bool to enum. I'm ready to push into github if including ggml.h in common.h is a fundamentally acceptable change.

Reply:
`common.h` includes `llama.h`, which includes `ggml.h`. It is safe to assume that `ggml.h` is already included in all the llama code.

Reply:
True enough. I didn't need that explicit include. Thanks!

Reply:
OK I've pushed those changes up as well as removing the mirror mode #ifdefs until the code to make them work is complete

Reply:
@bmtwl ^

Reply:
removed llama_numa_strategies and committed changes